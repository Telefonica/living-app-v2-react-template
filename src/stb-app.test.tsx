import { render } from '@testing-library/react';
import StbApp from './stb-app';

describe('<StbApp />', () => {
  beforeEach(() => {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (window as any).laSdk = {
      // TODO: LIVING-APP SDK TEAM
      // expose a mocking library for laSdk + Jest
      // we'll notify you as soon as this is available
      focus: jest.fn(),
      createSuggestions: jest.fn(),
      video: {
        start: jest.fn(),
        stop: jest.fn(),
      },
    };
  });

  test('render: match snapshot', () => {
    const { asFragment } = render(<StbApp />);
    expect(asFragment()).toMatchSnapshot();
  });
});
