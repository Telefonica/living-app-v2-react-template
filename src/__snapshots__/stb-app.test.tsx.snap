// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StbApp /> render: match snapshot 1`] = `
<DocumentFragment>
  <div
    class="stb-app"
  >
    <img
      class="stb-app__image"
      src=""
    />
    <div
      class="stb-app__text la-navigable"
      data-down="la-suggestions"
      id="center-elem"
    >
      Edit 
      <code>
        src/stb-app.tsx
      </code>
      .
    </div>
  </div>
</DocumentFragment>
`;
